File: C/src/static_list.c
#include <stdio.h>
#include "static_list.h"

void inicializarStaticList(StaticList *l) {
    l->tamanho = 0;
}

int inserirFinalStatic(StaticList *l, Registro r) {
    if (l->tamanho >= MAXTAM) return 0;

    l->itens[l->tamanho++] = r;
    return 1;
}

int removerValorStatic(StaticList *l, int valor) {
    for (int i = 0; i < l->tamanho; i++) {
        if (l->itens[i].valor == valor) {
            for (int j = i; j < l->tamanho - 1; j++) {
                l->itens[j] = l->itens[j + 1];
            }
            l->tamanho--;
            return 1;
        }
    }
    return 0;
}

void imprimirStaticList(StaticList *l) {
    for (int i = 0; i < l->tamanho; i++) {
        printf("%d ", l->itens[i].valor);
    }
    printf("\n");
}

void salvarStaticListEmArquivo(StaticList *l, const char *nomeArquivo) {
    FILE *arquivo = fopen(nomeArquivo, "w");
    if (arquivo == NULL) {
        perror("Erro ao abrir o arquivo para escrita");
        return;
    }

    for (int i = 0; i < l->tamanho; i++) {
        fprintf(arquivo, "%d \n", l->itens[i].valor);
    }

    fclose(arquivo);
}

File: Java/src/StaticList.java
public class StaticList {
    protected Registro[] elementos;
    protected int tamanho;
    protected int capacidade;

    public StaticList(int capacidade) {
        this.capacidade = capacidade;
        this.elementos = new Registro[capacidade];
        this.tamanho = 0;
    }

    public boolean inserirFim(Registro valor) {
        if (this.tamanho < this.capacidade) {
            this.elementos[this.tamanho] = valor;
            this.tamanho++;
            return true;
        }
        return false; // Lista cheia
    }

    public Registro removerInicio() {
        if (this.tamanho == 0) return null; // Lista vazia

        Registro valorRemovido = this.elementos[0];
        for (int i = 0; i < this.tamanho - 1; i++) {
            this.elementos[i] = this.elementos[i + 1];
        }
        this.tamanho--;
        return valorRemovido;
    }

    public void imprimir() {
        for (int i = 0; i < this.tamanho; i++) {
            System.out.print(this.elementos[i].valor + " ");
        }
        System.out.println();
    }

    public void salvarEmArquivo(String nomeArquivo) {
        try (java.io.FileWriter writer = new java.io.FileWriter(nomeArquivo)) {
            System.out.println("Salvando em " + nomeArquivo);
            for (int i = 0; i < this.tamanho; i++) {
                writer.write(this.elementos[i].valor + "\n");
            }
        } catch (java.io.IOException e) {
            System.err.println("Erro ao abrir o arquivo para escrita: " + e.getMessage());
        }
    }
}

